python/cudf/cudf/core/column/numerical.py

```diff
@@ -832,7 +832,8 @@ def can_cast_safely(self, to_dtype: DtypeObj) -> bool:
                     finfo = np.finfo(to_dtype_numpy)
                     lower_: int | float
                     upper_: int | float
-                    lower_, upper_ = finfo.min, finfo.max  # type: ignore[assignment]
+                    # assignment ignore not needed for numpy>=2.4.0
+                    lower_, upper_ = finfo.min, finfo.max  # type: ignore[assignment,unused-ignore]
```

Comment:
The `ignore[assignment]` can be removed once we're fully on numpy==2.4.0. For now, we need to ignore the ignore :)